game_logic.cpp

```diff
@@ -100,15 +130,17 @@ void guess(Board &board, Player &player) {
 		}
 	}
 	disablePosition(board, guess);
-	return;
 }
 
 // function to be called after each player has placed their mines
 // colliding mines are annihilated and removed from their owner's mine pool
 bool checkMineCollision(Board &board) {
 	bool wasThereCollision = false;
 	std::vector<Mine> conflictingMines;
-	for (int i = 0; i < board.placedMines.size() - 1; i++) { // size is -- because the vector value in i is compared to the other values on his right
+	for (int i = 0; i < board.placedMines.size() - 1; i++) { // size is -- because the vector value in i is compared to the values on its right
+		if (board.placedMines.size() <= 0) {

```

Comment:
#should use `.empty()` instead - `.size()` cannot return a negative value because it returns an `unsigned` one